#! /bin/sh /usr/share/dpatch/dpatch-run
## 34_debian_fixedscripts.dpatch by Petter Reinholdtsen

Provide default sequence numbers for a few well known scripts, to make
the generated boot sequence more closely match the original one.

It is not obvious that this is a good idea.  It might make the boot
sequence less optimal (as in less scripts can run in parallel), and
might affect the runtime complexity.  The rationale for doing this is
that it might make the boot sequence more familiar for those that knew
the old sequence, and that Petter Reinholdtsen claim to have seen some
Debian documentation stating where in the boot sequence some scripts
are expected to be located.  The document referense is yet to be
recovered.

@DPATCH@
--- insserv-1.09.0.orig/insserv.c
+++ insserv-1.09.0/insserv.c
@@ -2110,6 +2110,16 @@
     setorder("boot.setup", 20, false);
 
     /*
+     * Debian scripts with well known sequence numbers.  Not sure if
+     * we want to fix all of these.
+     */
+    setorder("checkroot.sh",	10, false); setlsb("checkroot.sh");
+    setorder("checkfs.sh",	30, false); setlsb("checkfs.sh");
+    setorder("networking",	40, false); setlsb("networking.sh");
+    setorder("mountnfs.sh",	45, false); setlsb("mountnfs.sh");
+    setorder("single",		90, false); setlsb("single");
+
+    /*
      * Set virtual dependencies for already enabled none LSB scripts.
      */
     nonlsb_script();
